python3Packages.urllib3: remove obsolete SSL dependencies

veprbl opened the pull request:
Same as
https://github.com/gentoo/gentoo/pull/19383

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Drop build time dependency on Rust

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

dotlambda commented:
/rebase staging

dotlambda commented on pkgs/development/python-modules/urllib3/default.nix:
```diff
@@ -23,7 +23,8 @@ buildPythonPackage rec {
 
   checkInputs = [ pytest mock tornado ];
-  propagatedBuildInputs = [ pyopenssl cryptography idna certifi ipaddress pysocks ];
+  propagatedBuildInputs = [ ipaddress pysocks ]
```
```suggestion
  propagatedBuildInputs = [ pysocks ]
```
https://github.com/urllib3/urllib3/blob/1.26.3/setup.py#L93

veprbl replied:
Good catch. Thanks!

dotlambda commented on pkgs/development/python-modules/urllib3/default.nix:
```diff
@@ -23,7 +23,8 @@ buildPythonPackage rec {
   checkInputs = [ pytest mock tornado ];
-  propagatedBuildInputs = [ pyopenssl cryptography idna certifi ipaddress pysocks ];
+  propagatedBuildInputs = [ ipaddress pysocks ]
+    ++ lib.optionals isPy27 [ pyopenssl cryptography idna certifi ];
```
```suggestion
    ++ lib.optionals isPy27 [ pyopenssl cryptography idna certifi ipaddress ];
```

dotlambda commented on pkgs/development/python-modules/urllib3/default.nix:
```diff
@@ -23,7 +23,8 @@ buildPythonPackage rec {
   doCheck = false;
 
   checkInputs = [ pytest mock tornado ];
-  propagatedBuildInputs = [ pyopenssl cryptography idna certifi ipaddress pysocks ];
```
Do we really want to remove `certifi`? That's not used for Python 2 support, but for having a list of root certificates. I don't know if we bake our own root certificates into Python.

veprbl replied:
Actually, I don't see it imported at all. I see it used to be referenced in the documentation at some point.

https://github.com/urllib3/urllib3/commit/90ee766d9153e2f0def7ba903895d51a86c13b5f

> We install certifi on every version of Python because it's awesome

dotlambda replied:
Indeed. So let's just remove it completely.

veprbl committed:
python2Packages.urllib3: re-enable

Was disabled for no explained reason.

Fixes: dced68b7 ('python3Packages.urllib3: enable tests (#114282)')

veprbl committed:
python3Packages.urllib3: remove obsolete SSL dependencies

Same as:
'https://github.com/gentoo/gentoo/pull/19383'

Remove certifi as it is not referenced
https://github.com/urllib3/urllib3/commit/90ee766d9153e2f0def7ba903895d51a86c13b5f
